docs/structured-streaming-programming-guide.md

```diff
@@ -65,11 +51,13 @@ val words = lines.as[String].flatMap(_.split(" "))
 val wordCounts = words.groupBy("value").count()
 {% endhighlight %}
 
-This `lines` DataFrame represents an unbounded table containing the streaming text data. This table contains one column of strings named “value”, and each line in the streaming text data becomes a row in the table. Note, that this is not currently receiving any data as we are just setting up the transformation, and have not yet started it. Next, we have converted the DataFrame to a  Dataset of String using `.as(Encoders.STRING())`, so that we can apply the `flatMap` operation to split each line into multiple words. The resultant `words` Dataset contains all the words. Finally, we have defined the `wordCounts` DataFrame by grouping by the unique values in the Dataset and counting them. Note that this is a streaming DataFrame which represents the running word counts of the stream.
```

Comment:
Scala snippet uses `.as[String]` while Java's uses `.as(Encoders.STRING())`.
In Scala context, it is weird to read "we have converted the DataFrame to a  Dataset of String using `.as(Encoders.STRING())`" while *we* actually used `.as[String]`